src/main.cpp

```diff
@@ -6005,6 +6005,9 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
         {
             boost::this_thread::interruption_point();
 
+            if (!(addr.nServices & NODE_NETWORK))
+                continue;
+
```

Comment:
i'm not sure this is such a good idea...if we don't share pruned nodes (non node-network) then we'll never try to connect to them, if we never connect with them then they'lll never get new addresses, and no one will ever know about them and they won't know about us.?  Isn't this possibly a reason why we don't seem to be getting as many incoming connections as they used to?